superduperdb/models/langchain/retriever.py

```
@@ -31,29 +31,31 @@ def __init__(
         self.key = key
         self.n = n
 
-    def get_relevant_documents(self, query: str) -> List[Document]:
+    def get_relevant_documents(self, query: str) -> t.List[Document]:
         document_to_search = {self.key: query}
         ids = self.vector_index.get_nearest(
-            document_to_search,
+            document_to_search,  # type: ignore
```

One of the points of adding `typing` is to find issues and errors in the code.

In this case, because both the variable and the function call are typed, this indicates an actual issue in the code. So suppressing the error isn't the right solution.

`document_to_search` is a `dict` but the type of the first parameter to `vector_index.get_nearest` is `Document`.  This code path has probably never been executed.

----

This needs an actual code fix!  

Consider releasing it as a separate commit before the typing commit.

---

The variable should be a `Document` but even the function seems to mix `langchain.schema.Document` and `superduperdb.core.documents.Document`. This does not seem like something that a simple fix would solve.

---

> even the function seems to mix langchain.schema.Document and superduperdb.core.documents.Document. 

Here's the function: https://github.com/SuperDuperDB/superduperdb-stealth/blob/15554c9644b04d8f67800e0ddab76c044a73a0d5/superduperdb/models/langchain/retriever.py#L34-L56

In fact, it only uses one type of Document, `langchain.schema.Document`.  

As discussed, a correct fix to the code would be this:

at the top:
```
from superduperdb.core import documents
```

On line 35:

```
         document_to_search = documents.Document({self.key: query})
````

----

I have never seen any of this code before; neither did I use an IDE; I just learned all of this by reading the code during this review.

---

Changed.